server/jetstream_test.go

```diff
@@ -11605,25 +11605,25 @@ func TestJetStreamMirrorBasics(t *testing.T) {
 			si, err := js2.StreamInfo(streamName, &nats.StreamInfoRequest{SubjectsFilter: subject})
 			require_NoError(t, err)
 			if ss, ok := si.State.Subjects[subject]; !ok {
-				t.Logf("Expected messages with the transformed subject %s", subject)
+				return fmt.Errorf("expected messages with the transformed subject %s", subject)
 			} else {
-				if ss != subjectNumMsgs {
-					t.Fatalf("Expected %d messages on the transformed subject %s but got %d", subjectNumMsgs, subject, ss)
+				if ss > subjectNumMsgs || ss < subjectNumMsgs {
```

Comment:
This is just `ss !- subjectNumMsgs` like it was originally.

Reply:
Yes but it now properly returns an error (it wasn't using checkFor correctly by not returning an error which caused checkFor to stop trying again).

Reply:
Yes saw that, but the if is non-sensical, its just `!=`

Reply:
Changed